modules/ROOT/pages/salesforce/salesforce-connector-reference-99.adoc

```diff
@@ -0,0 +1,5927 @@
+* SALESFORCE:RETRY_EXHAUSTED
+* SALESFORCE:UNKNOWN
+
+
+[[CloseJob]]
+== Close Job
+`<salesforce:close-job>`
+
+
+Closes an open Job given its ID.
+
+
+=== Parameters
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Job id a| String |  The Job ID identifying the Job to be closed. |  | x
+| Target Variable a| String |  The name of a variable that stores the operation's output |  |
+| Target Value a| String |  An expression that evaluates the operation's output and stores the output in Target Variable |  #[payload] |
+| Reconnection Strategy a| * <<reconnect>>
+* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
+|===
+
+=== Output
+
+[%autowidth.spread]
+|===
+| *Type* a| <<JobInfoType>>
+|===
+
+=== For Configurations
+
+* <<SfdcConfig>>
+
+=== Throws
+
+* SALESFORCE:CONNECTIVITY
+* SALESFORCE:INVALID_REQUEST_DATA
+* SALESFORCE:INVALID_SESSION
+* SALESFORCE:INVALID_STRUCTURE_FOR_INPUT_DATA
+* SALESFORCE:RETRY_EXHAUSTED
+* SALESFORCE:UNKNOWN
+
+
+[[ConvertLead]]
+== Convert Lead
+`<salesforce:convert-lead>`
+
+
+Converts a Lead into an Account, Contact, or (optionally) an Opportunity.
+
+
+=== Parameters
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | The name of the configuration to use. | | x
+| Lead Convert Request a| <<LeadConvertRequest>> |  Information needed for lead conversion |  #[payload] |
+| Headers a| Object |  |  |
+| Target Variable a| String |  The name of a variable that stores the operation's output |  |
+| Target Value a| String |  An expression that evaluates the operation's output and stores the output in Target Variable |  #[payload] |
+| Reconnection Strategy a| * <<reconnect>>
+* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
+|===
+
+=== Output
+
+[%autowidth.spread]
+|===
+| *Type* a| <<LeadConvertResult>>
+|===
+
+=== For Configurations
+
+* <<SfdcConfig>>
+
+=== Throws
+
+* SALESFORCE:CONNECTIVITY
+* SALESFORCE:INVALID_REQUEST_DATA
+* SALESFORCE:INVALID_SESSION
+* SALESFORCE:INVALID_STRUCTURE_FOR_INPUT_DATA
+* SALESFORCE:RETRY_EXHAUSTED
+* SALESFORCE:UNKNOWN
+
+
+[[Create]]
+== Create
+`<salesforce:create>`
+
+
+Adds one or more new records to your organization's data. When you map your objects to the input of this message processor keep in mind that they need to match the expected type of the object at Salesforce. Take the CloseDate of an Opportunity as an example, if you set that field to a string of value "2019-12-13" it will be sent to Salesforce as a string and operation will be rejected on the basis that CloseDate is not of the expected type. The proper way to actually map it is to generate a Java Date object, you can do so using Groovy expression evaluator as `#[groovy:Date.parse("yyyy-MM-dd", "2019-12-13")]`.
```

Comment:
```suggestion
Adds one or more new records to your organization's data. When you map your objects to the input of this message processor keep in mind that they need to match the expected type of the object at Salesforce. 

For example, if you set the CloseDate field of an Opportunity to a string of value of "2019-12-13" it will be sent to Salesforce as a string and the operation will be rejected on the basis that CloseDate is not of the expected type. The proper way to map it is to generate a Java Date object, which you can do  by using a Groovy expression evaluator as `#[groovy:Date.parse("yyyy-MM-dd", "2019-12-13")]`.
```